abci/types/types.go

```diff
@@ -167,3 +170,125 @@ func RespondVerifyVoteExtension(ok bool) ResponseVerifyVoteExtension {
 		Result: result,
 	}
 }
+
+func MustHashResults(r []*ExecTxResult) []byte {
```

Comment:
```suggestion
// MustHashResults encodes the given results and computes a Merkle hash
// of their binary contents. It will panic if encoding fails.
func MustHashResults(r []*ExecTxResult) []byte {
```

Reply:
At a higher level, I wonder if we shouldn't make most of these functions accept byte slices directly, and make the caller handle encoding. It seems like the only way these things could panic is if marshaling fails. Perhaps we can provide a "MarshalToByteSlices" function that reports an error, and make the caller handle that error. Then these hashing helpers have no possible mode of failure.

While encoding "shouldn't" fail, it can, and I'm not sure we're gaining much value from threading that responsibility through multiple levels. Something to consider, anyway.

Reply:
Done. This allowed for a few nice little cleanups anyway.